k8s/chart/templates/rbac.yaml

```diff
@@ -0,0 +1,56 @@
+{{ if .Values.rbac.create}}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: {{ include "sdp-operator.name" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "sdp-operator.labels" . | nindent 4 }}
+  annotations:
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-weight": "1"
+rules:
+  - apiGroups: [ "{{ .Values.operator.version }}.appgate.com"]
+    resources:
+      - policies
+      - entitlements
+      - conditions
+      - ringfencerules
+      - appliances
+      - sites
+      - ippools
+      - identityproviders
+      - localusers
+      - administrativeroles
+      - mfaproviders
+      - adminmfasettingss
+      - trustedcertificates
+      - criteriascriptss
+      - devicescripts
+      - entitlementscripts
+      - appliancecustomizations
+      - globalsettingss
+      - clientconnections
+    verbs: ["get", "watch", "list"]
+  - apiGroups: [""]
+    resources: ["configmaps", "secrets"]
+    verbs: ["get", "watch", "list", "update", "patch"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: {{ include "sdp-operator.name" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+  {{- include "sdp-operator.labels" . | nindent 4 }}
+  annotations:
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-weight": "-1"
```

Comment:
Same here. Dont need hook annotation.

If you are worried about how the resources gets created on a cluster, helm will take care that for you. After rendering, Helm will apply the resources in the following order: https://github.com/helm/helm/blob/ac925eb7279f4a6955df663a0128044a8a6b7593/pkg/releaseutil/kind_sorter.go#L27